scripts/core/editor3/helpers/highlights.js

```diff
@@ -444,41 +574,110 @@ export function getRangeAndTextForStyle(editorState, style) {
         throw new Error('Only collapsed selection supported');
     }
 
-    var blockKey = selection.getAnchorKey();
-    var block = editorState.getCurrentContent().getBlockForKey(blockKey);
-    var characterLists = block.getCharacterList();
-
-    var from = null;
-    var to = null;
+    const content = editorState.getCurrentContent();
```

Comment:
Seems like the function got too hairy, `eslint` complexity measurement is finally correct :smile: 

Why didn't it work as it was? Can you simplify it? I think removing the mutations would help.